### PR TITLE
Rotation, Dynamic Z-Index and a bugfix

### PR DESCRIPTION
**Pull request by PixelRifts**

### CODE REVIEW

**Comment by PixelRifts** [comment]
Rotation has the side effect of shifting all sprites so the colliders won't line up correctly. Will wait until Juyas's collision system gets merged to implement a fix for the colliders

**Review by athaun — APPROVED** [review]
This looks great, I am assuming you plan on fixing rotation for the debug renderer and collisions after Juyas's PR is merged.
I like the centered position for the gameObjects, although it may be good to allow the user specify where they want the origin if they don't like the center for whatever they are doing.

**Pull request merged**
